packages/core/src/utils/public.ts

```diff
@@ -321,3 +322,23 @@ export function shuffle(array: any[]): void {
 export function generateId(): string {
   return nanoid();
 }
+
+/**
+ * parse plugin name, return the plugin protocol and url object
+ */
+export function parsePluginName(name: string): { protocol: PluginProtocol; urlObject: url.UrlWithStringQuery } {
+  const urlObject = url.parse(name);
+  let protocol: PluginProtocol;
+  if (path.isAbsolute(name)) {
+    protocol = 'fs';
+  } else if (/^git(\+ssh|\+https|\+http)?:$/.test(urlObject.protocol)) {
+    protocol = 'git';
+  } else if ([ 'https:', 'http:' ].indexOf(urlObject.protocol) !== -1) {
```

Comment:
This doesn't mean tarbar, just for http/https url.

Reply:
For plugin, it's tarball.

Reply:
I think we need to verify if it ends with .tgz or .gz, just checking for protocol is not enough in this case.

Reply:
OK, it's better to check the path.